inst/include/sampler.h

```diff
@@ -332,10 +431,12 @@ namespace smc {
         //Move the particle set.
         MoveParticles();
 
-        //Normalise the weights to sensible values....
-        double dMaxWeight = arma::max(pPopulation.GetLogWeight());
-        pPopulation.SetLogWeight(pPopulation.GetLogWeight() - (dMaxWeight)*arma::ones(N));
+        //Estimate the normalising constant
+        dlogNCIt = CalcLogNC();
+        dlogNCPath += dlogNCIt;
 
+        //Normalise the weights
+        pPopulation.SetLogWeight(pPopulation.GetLogWeight()  - dlogNCIt*arma::ones(N));
```

Comment:
Do you actually need to create the vector of ones here (glancing at the Armadillo docs, isn't arma::vec - double well defined)? If you do need the extra vector, it might be sensible to have a static N-vector of ones available to avoid the overheads.